src/main/java/sg/ncl/DataController.java

```diff
@@ -188,6 +205,129 @@ public String getPublicDatasets(Model model) {
+            resumableObject.put("resumableFilename", info.resumableFilename);
+            resumableObject.put("resumableRelativePath", info.resumableRelativePath);
+
+            String resumableChunk = Base64.encodeBase64String(os.toByteArray());
+            log.debug(resumableChunk);
+            resumableObject.put("resumableChunk", resumableChunk);
+
+            String url = properties.sendUploadChunk(datasetId, resumableChunkNumber);
+            log.debug("URL: {}", url);
+            HttpEntity<String> httpEntity = createHttpEntityWithBody(resumableObject.toString());
+            restTemplate.setErrorHandler(new MyResponseErrorHandler());
+            ResponseEntity responseEntity = restTemplate.exchange(url, HttpMethod.POST, httpEntity, String.class);
+            String body = responseEntity.getBody().toString();
+
+            if (RestUtil.isError(responseEntity.getStatusCode())) {
+                throw new Exception();
+            }
+            return ResponseEntity.ok(body);
+        } catch (Exception e) {
+            log.error("Error sending upload chunk for dataset " + datasetId);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Error sending upload chunk");
+        }
+    }
+
+    /**
+     * References:
+     * [1] http://stackoverflow.com/questions/25854077/calling-a-servlet-from-another-servlet-after-the-request-dispatcher-forward-meth
+     * [2] http://stackoverflow.com/questions/29712554/how-to-download-a-file-using-a-java-rest-service-and-a-data-stream
+     * [3] http://stackoverflow.com/questions/32988370/download-large-file-from-server-using-rest-template-java-spring-mvc
+     */
+    @RequestMapping(value="{datasetId}/resources/{resourceId}", method=RequestMethod.GET)
+    public void getResource(@PathVariable String datasetId,
+                            @PathVariable String resourceId,
+                            final HttpServletResponse httpResponse) throws UnsupportedEncodingException {
+        try {
+            // Optional Accept header
+            RequestCallback requestCallback = request -> {
+                request.getHeaders().setAccept(Arrays.asList(MediaType.APPLICATION_OCTET_STREAM, MediaType.ALL));
+                request.getHeaders().set("Authorization", httpScopedSession.getAttribute(webProperties.getSessionJwtToken()).toString());
+            };
+
+            // Streams the response instead of loading it all in memory
+            ResponseExtractor<Void> responseExtractor = response -> {
+                // Here I write the response to a file but do what you like
+                String content = response.getHeaders().get("Content-Disposition").get(0);
+                Pattern pattern = Pattern.compile("'(.*?)'");
+                Matcher matcher = pattern.matcher(content);
+                if (matcher.find()) {
+                    Path path = Paths.get("some/path", matcher.group(1));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this unused "path" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1481)